Add Option for Open/Close Trigger

crysxd opened the pull request:
I added a simple setting to allow the user to decide whether the emergency stop should be triggered when the button is opened or closed. I only had buttons which close on press, so the emergency stop was instantly triggered (unless I kept the button pressed).

Commit:
text correction

LuckyX182 committed:
Update README.md

AliceGrey committed:
Fixed notification typos

Fixed notification typos/grammar errors. These were bothering me.

LuckyX182 committed:
Merge pull request #3 from AliceGrey/AliceGrey-patch-1

Fixed notification typos

Commit:
test button functionality

Commit:
Pin number range

jneilliii committed:
Update __init__.py

LuckyX182 committed:
Merge pull request #4 from jneilliii/test_button

Update __init__.py

Commit:
integer fix

Commit:
test sensor result reset after settings hide

Commit:
Rest call error handling

Commit:
test pin logic working

Commit:
Already used pins shown - test button

Commit:
Changed bounce time

Commit:
changed some texts

Commit:
pin validation before settings save

Commit:
text change

Commit:
Documentation update

LuckyX182 committed:
Update README.md

Commit:
screenshot crop

Commit:
save fix

Commit:
Merge branch 'test_button'

# Conflicts:
#	octoprint_filamentsensorsimplified/templates/filamentsensorsimplified_settings.jinja2

Commit:
M113 fix

Commit:
merge fix

Commit:
changed event to work on older OctoPrint

Commit:
GPIO setup fix

Commit:
GPIO setup fix 2

alexiri commented:
This fails because there's no default value defined in `get_settings_defaults`.

alexiri commented on octoprint_emergencystopsimplified/templates/emergencystopsimplified_settings.jinja2:
```diff
@@ -16,17 +16,21 @@
-        <div class="alert-info wy-alert-warning marginTop">
-            <span class="icon-info-sign"/>
-            <span class="octoFontFamily iconIndent">Button is triggered when open.</span>
+        <label class="control-label">{{ _('Emergency stoped is triggered when:') }}</label>
```
```suggestion
        <label class="control-label">{{ _('Emergency stop is triggered when:') }}</label>
```

Commit:
GPIO setup fix

Commit:
Code comments + custom gcode changes

Commit:
Code refactoring

Commit:
mode setting

Commit:
gpio mode fixes

Commit:
more gpio mode fixes

Commit:
more gpio mode fixes

Commit:
gpio mode test button fixes

Commit:
triggered type

Commit:
triggered type fix

Commit:
triggered type fix

Commit:
triggered type fix test button

Commit:
GUI cleanup

Commit:
Logic changes

Commit:
Logic changes

Commit:
testing

Commit:
testing 2

Commit:
disabling fields when printing

Commit:
disabling all fields when printing

Commit:
disabling all fields when printing

Commit:
disabling all fields when printing

Commit:
disabling all fields when printing done

Commit:
Documentation + small updates

Commit:
Documentation updates

Commit:
False positives fix

LuckyX182 committed:
Version bug fix

Commit:
GPIO mode fix

Commit:
GPIO mode fix

LuckyX182 committed:
version and description update

LazeMSS committed:
UI improvements
Fixed a couple of UI bugs and inline styles
Made javascript logic to show warning when setting up bad configs

LazeMSS committed:
Removed debug

LazeMSS committed:
Validation improvements
Using 0 as disabled - making validation and input easier
Fixed bug where one could change the pin to invalid if the board mode was not modified
Improved UI validation:
 - Now check against valid data pins when in board mode
 - Only allow pin 1-27 when in BCM mode
 - Added texts to describe Physical/Board and GPIO/BCM modes

LazeMSS committed:
Added better turn on/off state handling

LazeMSS committed:
Fixed typos and more

LazeMSS committed:
Fixed readme and added FA3 icons

LuckyX182 committed:
Merge pull request #21 from LazeMSS/UIImprovements

UI improvements

Commit:
Merge branch 'master' into develop

Commit:
screenshot and version update

Commit:
removed unnecessary cleanup and added small error log

Commit:
test button texts changed

Commit:
javascript fix

Commit:
icon fix

Commit:
double detection prevention

Commit:
test button fix

Commit:
test button logic improvements

Commit:
handle deliberate filament change

Commit:
version and readme update

mike-schiller committed:
Resetting a couple state variables to ensure that filament changes can happen (are repeatable) multiple times per print.

mike-schiller committed:
dropping debug statement and removing lines indicating what I'd changed

LazeMSS committed:
Added support for using the native octoprint pause function and therefore also a bit of UI changes

LazeMSS committed:
Quick hack - need to get the option from settings

LazeMSS committed:
Added basic navbar status icon

LazeMSS committed:
Added polling for filament status every 60 seconds

LazeMSS committed:
Fixed timing for icon

LazeMSS committed:
removed debug

LazeMSS committed:
Removed warning about missing M600 now defaults to native OP pause

LazeMSS committed:
Cleanup after review. Moved test result below button again. Changed navbar icons. Text changes

LuckyX182 committed:
Merge pull request #37 from mike-schiller/repeatable-changes

Repeatable changes

LuckyX182 committed:
Merge pull request #39 from LazeMSS/NavbarIcon

Native Octoprint pause + Navbar icon

crysxd committed:
Fire filament change event so other can react
